File: js/palindrome.js
exports.palindrome = function(word) {
    const word1 = String(word).toLocaleLowerCase()
    const cleaned = word1.replace(/[^a-zA-Z1-9]/gi,"")
    const indexable = cleaned.split("")
    
    
    let left = 0
    let right = cleaned.length -1

    for (let i = 0; i < indexable.length; i++) {
        if (indexable[left] === indexable[right]){
            left++
            right--
        } else {
            return false
        }
    }
    return true
};

// export {palindrome as palindrome}

// console.log(palindrome('racecar'))

File: python/palindrome.py
def palindrome(word):
    # clean the word from spaces and puctuation and lowercase it
    word_low = str(word).lower()
    word_cleaned = "".join(x for x in word_low if x.isalnum())

    # we keep 2 pointers moving in opposite directions during the length of the string, if any of them are not equal it is not a palindrome 
    left = 0
    right = -1
    for i in range(len(word_cleaned)):
        if word_cleaned[left] == word_cleaned[right]:
            left +=1
            right -=1
        else:
            return False
    return True